app/Controllers/indexController.php

```diff
@@ -194,7 +194,7 @@ public function opmlAction(): void {
 		$get = FreshRSS_Context::currentGet(true);
 		if (is_array($get)) {
 			$type = $get[0];
-			$id = $get[1];
+			$id = (int)($get[1]);
```

Comment:
```suggestion
			$id = (int)$get[1];
```

Reply:
Yes, it works without the parentheses, but I find the syntax dangerous as it requires to be sure of the priorities and I can better like to be explicit as soon as there is a series of connections (here, a variable and then an array index, but it could be more complex).

Reply:
I reassure you, there is no risk. The cast is done on everything on the right. In the same way if you use (object) on an array it is the whole array that is transformed into StdClass:: . If you use (object)$myArray['foo'] it is only the key foo that is cast.

Reply:
My point is that it requires not making mistakes with operator priorities, because it is not entirely true that it is everything on the right. See examples, which outputs two different things (quite trivial with the `+` operator, but still):

```php
<?php
$x = 1;
echo json_encode((string)$x + 1), "\n";
```

```php
<?php
$x = 1;
echo json_encode((string)($x + 1)), "\n";
```

But you are right that the current cases are very simple